`make deploy-kind` can fail sometime

`make deploy-kind` calls `./dev/tools/deploy-to-kube --image-prefix=kind.local/` which [calls](https://github.com/kubernetes-sigs/agent-sandbox/blob/main/dev/tools/deploy-to-kube#L70C7-L70C54) `os.walk`. 

`os.walk` calls `os.scandir` inside it which does not guarantee file order.

https://docs.python.org/3/library/os.html#os.scandir

```
...Return an iterator of [os.DirEntry](https://docs.python.org/3/library/os.html#os.DirEntry) objects corresponding to the entries in the directory given by path. The entries are yielded in arbitrary order, and the special entries '.' and '..' are not included....
```

This can lead to this error

```

#18 sending tarball 0.1s done
#18 DONE 0.2s
pushed image kind.local/agent-sandbox-controller:v20251106-v0.1.0-rc.1-2-g7034fda
Image: "kind.local/agent-sandbox-controller:v20251106-v0.1.0-rc.1-2-g7034fda" with ID "sha256:e2498408d19517621a88baf331ea0810ecd0b9b0c06f7707b3633aa85ba54e37" not yet present on node "agent-sandbox-control-plane", loading...
loaded image kind.local/agent-sandbox-controller:v20251106-v0.1.0-rc.1-2-g7034fda into kind cluster agent-sandbox
./dev/tools/deploy-to-kube --image-prefix=kind.local/
applying manifest k8s/rbac.generated.yaml
clusterrole.rbac.authorization.k8s.io/agent-sandbox-controller created
ko://sigs.k8s.io/agent-sandbox/cmd/agent-sandbox-controller: replaced image with kind.local/agent-sandbox-controller:v20251106-v0.1.0-rc.1-2-g7034fda
applying manifest k8s/extensions.controller.yaml
Error from server (NotFound): error when creating "STDIN": namespaces "agent-sandbox-system" not found
Traceback (most recent call last):
  File "/home/user/pleasework/agent-sandbox/./dev/tools/deploy-to-kube", line 112, in <module>
    main(args)
  File "/home/user/pleasework/agent-sandbox/./dev/tools/deploy-to-kube", line 101, in main
    subprocess.run(cmd, cwd=repo_root, check=True, input=modified_content, text=True)
  File "/usr/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['kubectl', 'apply', '-f', '-']' returned non-zero exit status 1.
make: *** [Makefile:18: deploy-kind] Error 1
```

Where `extensions.controller.yaml` is applied before `controller.yaml` and fails because namespace `agent-sandbox-control-plane` is not yet created.

--- a/dev/tools/deploy-to-kube
+++ b/dev/tools/deploy-to-kube
@@ -68,6 +68,10 @@ def main(args):
         return
 
     for root, dirs, files in os.walk(manifests_path):
+        # Process files in alphabetical order
+        # This works now because k8s/ doesn't contain CRs, and Namespace is defined in the first file (controller.yaml).
+        # TODO: Handle file ordering better so that CRDs and Namespaces are deployed first
+        files.sort()
         for filename in files:
             if not (filename.endswith(".yaml") or filename.endswith(".yml")):
                 continue

